src/app.js

```diff
@@ -1,26 +1,76 @@
 const getJobs = require("./getJobs");
+const getToken = require("./getToken");
 const express = require("express");
 const app = express();
-const port = 8081;
+const config = require("./config");
+const { port, origin, AppID, APP_SECRET, redirect_url, cookieAge } = config;
 const path = require("path");
+var cookieParser = require("cookie-parser");
 
 app.use(express.static(path.join(__dirname)));
+app.set("views", __dirname + "/views");
 
 app.set("view engine", "ejs");
 
-//app.get("redirect", (req,res) => res.render("pages/redirect"))
+app.use(cookieParser());
 
-app.get("/groups/:id/jobs", function (req, res) {
-  getJobs.getJobs(req.params.id).then((data) =>
-    res.render("pages/index", {
-      created: data.filter((data) => data.status === "created"),
-      pending: data.filter((data) => data.status === "pending"),
-      running: data.filter((data) => data.status === "running"),
-    })
+app.get("/redirect/:state", (req, res) => {
+  res.redirect(
+    `https://gitlab.com/oauth/authorize?` +
+      `&client_id=${AppID}` +
+      `&redirect_uri=${origin}/redirect` +
+      `&response_type=code` +
+      `&state=${req.params.state}` +
+      `&scope=profile+read_api`
   );
-  // .then((data) => console.log((data[0]).map(({status, ...rest})=> [status[1], rest])));
+});
+
+app.get("/redirect", (req, res) => {
+  code = req.query.code;
+  state = req.query.state;
+  getToken
+    .getToken(AppID, code, APP_SECRET, redirect_url)
+    .then((response) => {
+      res.cookie(`access_token`, response.access_token, { maxAge: cookieAge });
+      res.redirect(`${origin}${state}`);
+    })
+    .catch((err) =>
+      res.render("pages/error", {
+        error:
+          `Cannot obtain access token. ` +
+          `Please make sure Application ID, Application Secret, Redirect URL are valid`,
+      })
+    );
+});
+
+app.get("/groups/:id/jobs", function (req, res) {
+  if (
+    req.cookies.access_token === "" ||
+    req.cookies.access_token === undefined
```

Comment:
Use `== null` instead of `=== undefined`. This is a specific way on JS to check for null and undefined at the same time.